src/tpu_forwarder.rs

```diff
@@ -36,24 +38,37 @@ impl TpuForwarder {
                 "tpu_forwarder-msgs_received",
                 msgs.read().unwrap().packets.len()
             );
-
-            if let Some(leader_data) = cluster_info
+            let leader_data = cluster_info
                 .read()
                 .expect("cluster_info.read() in TpuForwarder::forward()")
                 .leader_data()
-                .cloned()
-            {
-                if leader_data.id == my_id || !ContactInfo::is_valid_address(&leader_data.tpu) {
+                .cloned();
+            if TpuForwarder::update_addrs(leader_data, &my_id, &msgs.clone()) {
+                msgs.read().unwrap().send_to(&socket)?
+            }
+            continue;
+        }
+    }
+
+    fn update_addrs(
+        leader_data: Option<ContactInfo>,
+        my_id: &Pubkey,
+        msgs: &Arc<RwLock<Packets>>,
+    ) -> bool {
```

Comment:
That returned bool seems arbitrary. Is it an error case?

Reply:
Yeah, not very rust like. I wasn't a 100% on which way to go between result and bool and it seemed pretty minor so I stuck with bool. I'll update it to a result.